### ISSUE TITLE
Are cython_args available?

### ISSUE THREAD

**Issue by noDGodyaev** [mesonbuild/meson#9018]
I'm trying to control Cython behavior during compilation, specifically `--cplus` flag to generate `.cpp` files instead of `.c`.

In `meson < 0.59` i made:
```meson 
_bar_cpp = custom_target('bar_cpp',
  output : 'bar.cpp',
  input : 'bat.pyx',
  command : [cython, '--cplus', '-3', '--fast-fail', '@INPUT@', '-o', '@OUTPUT@']
)

py3.extension_module(
  'bar', _bar_cpp,
  dependencies : [py3_dep],
)
```
In `meson >= 0.59` i try to do:
```meson
py3.extension_module(
  'bar', 'bar.pyx',
  dependencies : py3_dep,
  cython_args: '--cplus'
)
```
I get the `.c` generating files.
Console output:
>(venv) user:~/examples/build2$ meson compile
>[1/2] Compiling `C` object cy_ising.cpython-38-x86_64-linux-gnu.so.p/meson-generated_cy_ising.pyx.c.o

System:
* native build 
* Ubuntu 20.04, etc.
* Python== 3.8.0
* meson==0.59.99
* ninja==1.10.0

**Comment by dcbaker** [comment]
In general cython_args works, but currently .c is harcoded for the output name. I have a patch to add c++ output support

**Comment by eli-schwartz** [comment]
cython_args are documented as available, but they silently do nothing when you try them.

On the other hand, `add_project_arguments(..., language: 'cython')` does work???

For C++ support specifically, since #9017 and available in the latest version of meson, you can use `cython_language=cpp`, see https://mesonbuild.com/Release-notes-for-0-60-0.html#cython-can-now-transpile-to-c-as-an-intermediate-language

**Comment by dcbaker** [comment]
Wow, I just noticed that the slugification of that URL turns the meaning completely upside down 😮 

I'll look at why the `cython_args` in targets doesn't work.

**Comment by eli-schwartz** [comment]
Yeah, I lol'ed pretty hard at that one.

**Comment by tristan957** [comment]
This can be closed. @xclaesse PR got merged.

**Comment by eli-schwartz** [comment]
Fixed in #10173
